utils/wasm-builder/src/lib.rs

```diff
@@ -79,21 +90,57 @@ impl WasmBuilder {
                 .for_each(|cause| eprintln!("|      {cause}"));
             process::exit(1);
         }
+
+        WasmBuilder::force_build_script_rebuild_on_next_run();
     }
 
     fn build_project(mut self) -> Result<()> {
         // TODO: Check nightly toolchain
         self.wasm_project.generate()?;
+
         self.cargo
             .set_manifest_path(self.wasm_project.manifest_path());
         self.cargo.set_target_dir(self.wasm_project.target_dir());
-
         let profile = self.wasm_project.profile();
         let profile = if profile == "debug" { "dev" } else { profile };
         self.cargo.set_profile(profile.to_string());
+        self.cargo.set_features(&WasmBuilder::enabled_features(
+            &self.wasm_project,
+            &self.excluded_features,
+        )?);
+
         self.cargo.run()?;
         self.wasm_project.postprocess()
     }
+
+    /// Returns the features enabled for the current build.
+    fn enabled_features(
+        wasm_project: &WasmProject,
+        features_to_exclude: &[&str],
+    ) -> Result<Vec<String>> {
+        let features = env::vars()
+            .filter_map(|(key, _)| {
+                key.strip_prefix("CARGO_FEATURE_")
+                    .map(|feature| feature.to_lowercase())
+            })
+            .filter(|feature| {
+                // Omit the `default` feature because the generated project excludes it
+                feature != "default" && !features_to_exclude.contains(&feature.as_str())
+            })
+            .collect::<Vec<String>>();
+        wasm_project.match_features(&features)
+    }
+
+    /// Force cargo to rebuild the build script next time it is invoked (e.g. when a feature is added or removed).
+    fn force_build_script_rebuild_on_next_run() {
```

Comment:
The function name is too long.
```suggestion
    fn refresh_build_script() {
```

Reply:
I prefer a longer but more descriptive name if I can't come up with a shorter one as it seems more confusing